### DISCUSSION TITLE
Fill memory with default values

### DISCUSSION THREAD

**Discussion by Real-MJoe** [prb28/vscode-amiga-assembly#138] [Ideas]
Is it possible to pre-fill the memory with random values or with a pattern before start the emulatior?  Best for FS-UAE and WinUAE if feasible. :)

**Comment by prb28** [comment]
Doesn't look very difficult to implement with the write memory option... Just need a "generator" with a comprehensive syntax.

**Reply by prb28** [reply]
The tricky part is the "before emulation starts"... I thinks it is a request to winuae or fs-uae maintainers directly. If after the first breakpoint with a manual command is useful it's only a modification of the extension and it's more easy.

**Comment by phaze101** [comment]
@Real-MJoe why? I do not think this is a good idea. If someone creates an EMU with lots of memory like say the Vampire with AROS this will create a huge delay before the EMU will start. 

I personally run my dev environment with 8MB Chipmen and 512MB of RAM. I know it is not a real machine but I do not like running out of memory when I am trying things out. Then I have a different config with a standard A500 or A1200 to really test these.

**Comment by Real-MJoe** [comment]
Especially on real hardware, the memory is never preallocated with "0" after a reset. To simulate this behaviour, it would be nice if the EMU filled the free memory with random data. 
The background is to find errors when not everything is preallocated with "0".

**Comment by phaze101** [comment]
@Real-MJoe ok I see the point but I would prefer it as an option to run then to have it there always happening. I am thinking of not just the old Amigas but even future ones likes the Vampire. Also with an EMU you can do what you want. No one is stopping me from giving my Amiga 1GB of RAM. It is possible as you know. I want to use the same extension with the Vampire V4 since VASM supports it and that has more memory. 

There will always be a huge delay each time if this is setup before you start the emulator. Not sure I would want that or anyone else would. When you are developing you want speed and not having delay to start your environment. So, if one needs that then it is ok to enable it but having there all the time doesn't make sense to me.

**Reply by Real-MJoe** [reply]
Of course, it should be an option of the emulator. Also the " erase " should take over the emulator and not the extension. Because even 512MB are written in C++ and Co. very fast. 
The Extension is to be only the link between both.
